Fix pinned tab causing resize loop on adjacent tab

abulgit opened the pull request:
Closes #41467 

My first PR in Zed, any guidance or tips are appreciated.

This fixes the flickering/resize loop that occurred on the tab immediately to the right of a pinned tab.

Removed the conditional border on the pinned tabs container. The border was a visual indicator to show when unpinned tabs were scrolled, but it wasn't essential and was causing the layout thrashing.

Release Notes:

- Fixed

abulgit committed:
Fix layout thrashing

smitbarmase reviewed (changes requested):
I don't think this is the right way to handle this. This code is essential to add border when there are large amount of unpinned tabs and are scrolled.

Since this PR results in something like this:

<img width="270" height="62" alt="Image" src="https://github.com/user-attachments/assets/52accbca-2516-4a95-8235-d22acf02f663" />

I'm not able to reproduce this issue, but can you inspect `is_scrollable` and `is_scrolled` bool and figure which one of them is flipping on alternate renders?

We might have to go deeper in checking if `tab_bar_scroll_handle.offset().x` is flipping to `0` every once in a while or is it`tab_bar_scroll_handle.max_offset().width`.

abulgit committed:
Added scrollability threshold to privent flickering

abulgit commented:
Thank you so much @smitbarmase  for guiding me to tackle this issue. 
So I added some debug logging to see what was happening. Turns out both values were flipping, but the root cause was `max_offset` toggling between `0px` and `0.8px`.

Here's my observation from logs

When no border: `max_offset=0.8px` -> `is_scrollable=true` -> border gets added
When border added: available space shrinks -> `max_offset=0px` -> `is_scrollable=false` -> border removed ->
and it repeats forever

The 0.8px difference is smaller than the border width, so it creates a loop.

So I changed the scrollability check from `!max_scroll.is_zero()` to `max_scroll > px(2.0)`. This ignores tiny subpixel differences while still showing the border when tabs are actually scrollable. and it works!

smitbarmase committed:
compute only when needed

smitbarmase reviewed (approved):
Great, that’s a nice fix! I moved the code back to `children` since we should only run all these comparisons when necessary. Thanks for working on this.
